genesis/genesis.go

```diff
@@ -41,6 +41,31 @@ const (
 //
 // The ID of the new network is [networkID].
 
+// Genesis returns:
+// 1) The byte representation of the genesis state of the platform chain
+//    (ie the genesis state of the network)
+// 2) The asset ID of AVAX
+func Genesis(networkID uint32, filePath string) ([]byte, ids.ID, error) {
+	if len(filePath) == 0 || networkID == constants.MainnetID || networkID == constants.TestnetID {
+		return FromConfig(GetConfig(networkID))
+	}
+
+	customConfig, err := GetConfigFile(filePath)
+	if err != nil {
+		return nil, ids.ID{}, fmt.Errorf("unable to load provided genesis config at %s: %w", filePath, err)
+	}
+
+	if customConfig.NetworkID != networkID {
```

Comment:
I think we should also enforce that the number of validators at genesis is > 0

Reply:
Also, `StartTime` should be in the past and  `InitialStakeDuration` should be greater than 0. We should probably enforce that it's greater than some minimum. 15 Minutes, maybe?

Reply:
Ahh my bad. I, for some reason, assumed genesis files were already being validated within `FromConfig` but realize now this wasn't ever necessary because you couldn't provide a custom genesis 🤦 .